contracts/TxRelay.sol

```diff
@@ -0,0 +1,107 @@
+pragma solidity 0.4.15;
+
+
+// This contract is meant as a "singleton" forwarding contract.
+// Eventually, it will be able to forward any transaction to
+// Any contract that is built to accept it.
+contract TxRelay {
+
+    // Note: This is a local nonce.
+    // Different from the nonce defined w/in protocol.
+    mapping(address => uint) nonce;
+
+    // This mapping specifies a whitelist of allowed senders for transactions.
+    // There can be one whitelist per ethereum account, which is the owner of that
+    // whitelist. Users can specify which whitelist they want to use when signing
+    // a transaction. They can use their own whitelist, a whitelist belonging
+    // to another account, or skip using a whitelist by specifying the zero address.
+    mapping(address => mapping(address => bool)) public whitelist;
```

Comment:
I thought we had decided to go with a simple, single mapping here. But I guess it's fine. It's a nice simple elegant implementation.